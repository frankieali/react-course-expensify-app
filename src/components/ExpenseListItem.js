import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

const ExpenseListItem = ({id, description, amount, createdAt}) => (
  <div>
    <h3><Link to={`/edit/${id}`}>{description}</Link></h3>
    <p>amount: {amount} | created at: {moment(createdAt).format('MMM Do, YYYY')}</p>
  </div>
);

export default ExpenseListItem;