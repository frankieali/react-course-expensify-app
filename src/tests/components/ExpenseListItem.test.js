import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import { expenses } from '../fixtures/expenses';

test('render expense list item', () => {
  const wrapper = shallow(<ExpenseListItem {...expenses[1]} />)
  expect(wrapper).toMatchSnapshot();
});