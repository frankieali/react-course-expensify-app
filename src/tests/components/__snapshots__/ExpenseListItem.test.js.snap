// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render expense list item 1`] = `
<div>
  <h3>
    <Link
      replace={false}
      to="/edit/2"
    >
      Rent
    </Link>
  </h3>
  <p>
    amount: 
    $195.00
     | created at: 
    Dec 27th, 1969
  </p>
</div>
`;
